File: ios/RNViewRecorder.m
#import "React/RCTViewManager.h"
@interface RCT_EXTERN_MODULE(RNViewRecorder, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(imageBackgroundUri, NSString)
RCT_EXPORT_VIEW_PROPERTY(backgroundType, NSString)
RCT_EXPORT_VIEW_PROPERTY(backgroundColorString, NSString)


RCT_EXTERN_METHOD(setupRecorder:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(startRecording:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(stopRecording:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(pauseRecording:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(saveAsImage:(nonnull NSNumber *)node)


RCT_EXPORT_VIEW_PROPERTY(onRecorded, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onSetupDone, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onImageStored, RCTDirectEventBlock)

@end

File: ios/RNViewRecorder.swift
@objc(RNViewRecorder)
class RNViewRecorder: RCTViewManager {
    override func view() -> RecorderView! {
        let drawView = RecorderView();
        
//        let frameworkBundle = Bundle(for: DrawView.self)
//        guard let defaultLibrary = try? sharedDevice?.makeDefaultLibrary(bundle: frameworkBundle) else {
//            fatalError("Could not load default library from specified bundle")
//        }
        
        return drawView;
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc func setupRecorder(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! RecorderView                                       // 5
            component.setupRecorder()                          // 6
        }
    }
    
    @objc func startRecording(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! RecorderView                                       // 5
            component.startRecording()                          // 6
        }
    }
    
    @objc func stopRecording(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! RecorderView                                       // 5
            component.stopRecording()                          // 6
        }
    }

    @objc func pauseRecording(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! RecorderView                                       // 5
            component.pauseRecording()                          // 6
        }
    }
    
   
    
    //saveAsImage
    @objc func saveAsImage(_ node: NSNumber){
        DispatchQueue.main.async {                                // 2
            let component = self.bridge.uiManager.view(             // 3
                forReactTag: node                                     // 4
                ) as! RecorderView                                       // 5
            component.saveAsImage()                           // 6
        }
    }
}
